### PR TITLE
Async actor callback and promise

### PR DESCRIPTION
**Pull request by HarelM**
This PR is replacing callbacks with promises in the sources to allow easier maintenance.
Since source cache is using the promise versions now, there's no point in keeping the old callback API.

It also reverts the change from method to properties in the sources which caused an unwanted breaking change.
Also adds docs and type improvements.

Any input would be valuable.
Relevant issue: #3052

### CODE REVIEW

File: src/source/canvas_source.ts [hunk 1]
```diff
@@ -107,7 +107,7 @@ export class CanvasSource extends ImageSource {
         this.animate = options.animate !== undefined ? options.animate : true;
     }
 
-    load = () => {
+    async load() {
```
Comment (neodescis) [inline src/source/canvas_source.ts:110]: I don't see anything in this function that necessitates the async?

Comment (neodescis) [reply src/source/canvas_source.ts:110]: Ah, yep, that's it. ImageSource's load() is async. Should we just make all of the load() functions async for consistency then?

File: src/util/actor.test.ts [hunk 2]
```diff
@@ -64,7 +64,7 @@ describe('Actor', () => {
         expect(received).toBeFalsy();
     });
 
-    test('cancel a request of a cancelable registered callback will cancel it', async () => {
+    test('abort a request of will abort it', async () => {
```
Comment (neodescis) [inline src/util/actor.test.ts:67]:
```suggestion
    test('aborting a request will successfully abort it', async () => {
```

File: src/source/canvas_source.ts (hunk 1 above)
Comment (HarelM) [reply src/source/canvas_source.ts:110]: I think I did, I'll check.

Comment (HarelM) [reply src/source/canvas_source.ts:110]: Added in the last one. 
It's worth remembering that `load` is not part of the interface or the public API, but more of a common method that is used in all these classes. But, I like consistency, so why not? :-)

**Pull request merged**